### DISCUSSION TITLE
Oxygen concentration control

### DISCUSSION THREAD

**Discussion by HunterXHunter1997** [firemodels/fds#12858] [Q&A]
Hello all,
I want to decrease the Oxygen concentration in the fluid domain and see the impact on a Heptane pool fire.
Is that possible in Fds/Pyrosim?

**Comment by mcgratta** [comment]
Y_O2_INFTY

**Reply by HunterXHunter1997** [reply]
Hello Kevin, I am not familiar with FDS coding can you please explain more? I am using Pyrosim for my simulations.
BR

**Comment by mcgratta** [comment]
I am not too familiar with PyroSim. What you need to know is that PyroSim writes a text file with the suffix .fds. This file is read by FDS. You should add `Y_O2_INFTY=X` to the `MISC` line in the `.fds file.

**Reply by HunterXHunter1997** [reply]
Ok I will try it, Thanks!

**Comment by HunterXHunter1997** [comment]
Hello again,
is there any way to make the Oxygen concentration change during the time?
for example, I want the mass fraction to look like this:
(0-30)sec - 0.23Kg/Kg
(30-60)sec- 0.17Kg/Kg
(60-100)sec- 0.14Kg/Kg
So it is a function of time.
Is that possible in fds?

**Reply by mcgratta** [reply]
You would have to flush the domain with air that has its oxygen content gradually reduce. Create a SURF with MASS_FLUX's of N2 and O2 such that the O2 concentration decreases and N2 increases.
